infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -517,10 +517,10 @@ private void onClusterRoleChange(TopologyDescriptor newTopology) {
 
         // Stop ongoing replication, stopLogReplication() checks leadership and role as SOURCE
         // We do not update topology until we successfully stop log replication
-        if (topologyDescriptor.getLocalClusterDescriptor().getRole() == ClusterRole.SOURCE) {
+        if (isSource()) {
```

Comment:
once other models are supported, we will also have to limit the scope of this API to the federation use-case and not all source/sink clusters.  Just pointing out.  You can add a TODO V2 if you have time.

Reply:
umm, not really since its actually checking `!topology.getRemoteSinkClusters().isEmpty();` underneath.
but places where `isSource()` is used will change in the connectionModel PR

Reply:
no, I meant the roleChange api.  Role change is only valid for federation.

Reply:
yes, that restriction is there in the plugin implementation.